packages/opentelemetry-exporter-prometheus/src/prometheus.ts

```diff
@@ -0,0 +1,280 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExportResult } from '@opentelemetry/base';
+import { NoopLogger } from '@opentelemetry/core';
+import {
+  LabelValue,
+  MetricDescriptor,
+  MetricDescriptorType,
+  MetricExporter,
+  ReadableMetric,
+} from '@opentelemetry/metrics';
+import * as types from '@opentelemetry/types';
+import { createServer, IncomingMessage, Server, ServerResponse } from 'http';
+import * as Prometheus from 'prom-client';
+import * as url from 'url';
+import { ExporterConfig } from './export/types';
+
+export class PrometheusExporter implements MetricExporter {
+  static readonly DEFAULT_OPTIONS = {
+    port: 9464,
+    startServer: false,
+    endpoint: '/metrics',
+    prefix: '',
+  };
+
+  private readonly _registry = new Prometheus.Registry();
+  private readonly _logger: types.Logger;
+  private readonly _port: number;
+  private readonly _endpoint: string;
+  private _server: Server;
+  private readonly _prefix?: string;
+
+  // This will be required when histogram is implemented. Leaving here so it is not forgotten
+  // Histogram cannot have a label named 'le'
+  // private static readonly RESERVED_HISTOGRAM_LABEL = 'le';
+
+  /**
+   * Constructor
+   * @param config Exporter configuration
+   * @param callback Callback to be called after a server was started
+   */
+  constructor(config: ExporterConfig = {}, callback?: () => void) {
+    this._logger = config.logger || new NoopLogger();
+    this._port = config.port || PrometheusExporter.DEFAULT_OPTIONS.port;
+    this._prefix = config.prefix || PrometheusExporter.DEFAULT_OPTIONS.prefix;
+    this._server = createServer(this._requestHandler);
+
+    this._endpoint = (
+      config.endpoint || PrometheusExporter.DEFAULT_OPTIONS.endpoint
+    ).replace(/^([^/])/, '/$1');
+
+    if (config.startServer || PrometheusExporter.DEFAULT_OPTIONS.startServer) {
+      this.startServer(callback);
+    } else if (callback) {
+      callback();
+    }
+  }
+
+  /**
+   * Save current metric state so that it can be pulled by the metrics backend.
+   *
+   * @todo reach into metrics to pull metric values on endpoint
+   * In its current state, the exporter saves the current values of all metrics when export
+   * is called and returns them when the export endpoint is called. In the future, this should
+   * be a no-op and the exporter should reach into the metrics when the export endpoint is
+   * called. As there is currently no interface to do this, this is our only option.
+   *
+   * @param readableMetrics Metrics to be sent to the prometheus backend
+   * @param cb result callback to be called on finish
+   */
+  export(
+    readableMetrics: ReadableMetric[],
```

Comment:
Tests for `Counter` and `Gauge` export is missing, do you plan to add it later? This https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-exporter-prometheus/test/test-prometheus-stats.ts#L68-L90 might help you to add tests.

Reply:
I still need to add them, just got lost in the shuffle

Reply:
added tests

Reply:
Thanks for adding the tests, PR looks great now.